src/main/java/com/rcv/CommonDataFormatReader.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Ranked Choice Voting Universal Tabulator
+ * Copyright (c) 2019 Jonathan Moldover, Louis Eisenberg, and Hylton Edingfield
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+ * the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this
+ * program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Purpose:
+ * Helper class to read and parse a Common Data Format file into cast vote record objects
+ * and candidate names.
+ */
+
+package com.rcv;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import javafx.util.Pair;
+
+class CommonDataFormatReader {
+
+  // path of the source file
+  private final String filePath;
+
+  // function: CommonDataFormatReader
+  // purpose: class constructor
+  // param: filePath source file to read
+  CommonDataFormatReader(String filePath) {
+    this.filePath = filePath;
+  }
+
+  // function: getCandidates
+  // purpose: returns list of candidates parsed from CDF election json
+  Set<String> getCandidates()
+  {
+    // container for results
+    Set<String> candidates = new HashSet<>();
+    try {
+      // try to read in the file
+      HashMap json = JsonParser.readFromFile(filePath, HashMap.class);
```

Comment:
It could be, but that ends up producing Unckecked assignment warnings.  Without getting deeper into Java generics, this approach removed the warnings and works reasonably.